rope/base/oi/doa.py

```diff
@@ -118,11 +117,11 @@ class _SocketReceiver(_MessageReceiver):
     def __init__(self):
         self.server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         self.data_port = 3037
-        self.key = uuid.uuid4().bytes
+        self.key = bytes(os.urandom(32))
```

Comment:
`os.urandom` always returns bytes in all python versions

Reply:
```
>>> import os
>>> type(os.urandom(32))
<type 'str'>
```

ah, I forgot there's no distinction in python2. Done.